sympy/vector/vector.py

```diff
@@ -108,10 +108,12 @@ def dot(self, other):
         # Check if the other is a del operator
         if isinstance(other, Del):
             def directional_derivative(field):
-                field = express(field, other.system, variables=True)
-                out = self.dot(other._i) * df(field, other._x)
-                out += self.dot(other._j) * df(field, other._y)
-                out += self.dot(other._k) * df(field, other._z)
+                from sympy.vector.operators import _get_coord_sys_from_expr
+                coord_sys = _get_coord_sys_from_expr(field)
```

Comment:
This line introduce some problems, because, when object not from `CoordSysCartesian` is put, we have AttributeError:
```python
    out = self.dot(coord_sys._i) * df(field, coord_sys._x)
AttributeError: 'NoneType' object has no attribute '_i'
```
We can handle this scenario separately.